worker/sandbox/main.go

```diff
@@ -0,0 +1,60 @@
+package sandbox
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// Sandbox provides a way to run an arbitary command
+// within a sandbox, with configured input/outputs and
+// proper time and memory limits.
+//
+// kjudge currently implements two sandboxes, "isolate" (which requires "github.com/ioi/isolate" to be available)
+// and "raw" (NOT RECOMMENDED, RUN AT YOUR OWN RISK).
+// Which sandbox is used can be set at runtime with a command-line switch.
+type Sandbox interface {
+	Run(*SandboxInput) (*SandboxOutput, error)
+}
```

Comment:
`sandbox.Sandbox` is an annoying name to work with now. Perhaps something like `sandbox.Runner`? It would also match Go interface names better (`Runner` has a `Run` method)